### DISCUSSION TITLE
help me please

### DISCUSSION THREAD

**Discussion by Seungzu** [hyperledger/fabric-samples#1220] [General]
# 1. Build a network passively
- I am now trying to build test-network manually by entering commands directly without using network.sh.
```sh
# These two lines are commands provided in fabric docs.
# ./network.sh will not be used.
$ ./network.sh up createChannel -c mychannel -ca
$ ./network.sh deployCC -ccn basic -ccp ../asset-transfer-basic/chaincode-javascript/ -ccl javascript
```

# 2. Sequence of operations
- This is the task sequence for manual construction of test-network.
- Refer to the network.sh file below.
```sh
# ca container - orderer, org1, org2
$ docker-compose -f compose/compose-ca.yaml -f compose/docker/docker-compose-ca.yaml up -d

# Setting environment variables
export PATH=${PWD}/../bin:$PATH
export FABRIC_CFG_PATH=$PWD/../config/

# Generate certificate (orderer, org1, org2)
. organizations/fabric-ca/registerEnroll.sh
createOrg1
createOrg2
createOrderer

# Create connection.json file
./organizations/ccp-generate.sh
```
It worked fine up to this point.

# 3. next step & error
- The next task requires creating orderer, org1, and org2 containers.
- I think the command to create orderer, org1, and org2 containers is one of the two below.
- However, errors occur in both cases.
```sh
docker-compose -f compose/compose-test-net.yaml -f compose/docker/docker-compose-test-net.yaml up -d
# An error occurs as follows
# invalid spec: :/host/var/run/docker.sock: empty section between colons

docker-compose -f compose/compose-test-net.yaml up -d
# The orderer container was created normally.
# The org1 and org2 containers are output as “exited”.
# The logs of org1 and org2 containers are as follows.
# Fatal error when initializing core config: FABRIC_CFG_PATH /etc/hyperledger/peercfg does not exist
```

# 4. Please save me
- If you know a solution, please let me know.

**Comment by satota2** [comment]
@Seungzu 

> docker-compose -f compose/compose-test-net.yaml -f compose/docker/docker-compose-test-net.yaml up -d

This error seems to occur because the DOCKER_SOCK environment variable is not specified.
I think it will work if you set it as follows, as specified in test-network:

```bash
DOCKER_SOCK="/var/run/docker.sock" docker-compose -f compose/compose-test-net.yaml -f compose/docker/docker-compose-test-net.yaml up -d
```

**Reply by Seungzu** [reply]
DOCKER_SOCK="/var/run/docker.sock" docker-compose -f compose/compose-test-net.yaml -f compose/docker/docker-compose-test-net.yaml up -d

An error occurs when proceeding like this
```md
bash: /var/run/docker.sock: Permission denied
```

So I proceeded with “sudo” in front, and an error occurred.
```md
sudo: /var/run/docker.sock: command not found
```

**Reply by satota2** [reply]
@Seungzu 
I'm not certain about your environment, but adding the user to the docker group with the following command might resolve the issue.
https://hyperledger-fabric.readthedocs.io/en/latest/prereqs.html#linux-ubuntu-debian-based-distro

```bash
# Add your user to the Docker group.
sudo usermod -a -G docker <username>
```

**Reply by Seungzu** [reply]
```sh
# install docker (v26.1.4)
$ curl -fsSL https://get.docker.com/ | sudo sh
$ sudo usermod -aG docker $USER

# install docker-compose (v2.27.1)
$ mkdir /usr/local/bin
$ sudo curl -L "https://github.com/docker/compose/releases/download/v2.27.1/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local /bin/docker-compose
$ sudo chmod +x /usr/local/bin/docker-compose

# install fabric (v2.5.8)
curl -sSL https://bit.ly/2ysbOFE | bash -s -- 2.5.8 1.5.11
```
The above are installation tasks before building a fabric network.
Is there something wrong here?
